### DISCUSSION TITLE
Unable to create a local repository in GIT

### DISCUSSION THREAD

**Discussion by bharatbondada** [community/community#23382] [New to GitHub]
<p>Unable to create a local repository .git in the project folder.</p>
<p>.git folder is not visible in the folder.</p>
<p>Please support me with a solution.</p>

**Comment by Zerotask** [comment]
<p>Did you try: git init</p>
<p>Did that command show an error? If yes, please post the error.<br>
If you don’t see this folder, try to check the option to show hidden folders in your file explorer or try the command: ls -al</p>

**Comment by bharatbondada** [comment]
<p>Yes, I was using the command git init from the git bash.</p>
<p>Initialized empty Git repository in C:/ShreyasBKR_Project/.git/</p>
<p>But unable to view the .git folder.</p>
<p>I have checked for invisible folders also.</p>

**Comment by tajmone** [comment, marked as answer]
<aside class="quote no-group" data-username="bharatbondada" data-post="3" data-topic="190223" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 bharatbondada:</div>
<blockquote>
<p>Yes, I was using the command git init from the git bash.</p>
<p>Initialized empty Git repository in C:/ShreyasBKR_Project/.git/</p>
<p>But unable to view the .git folder.</p>
<p>I have checked for invisible folders also.</p>
</blockquote>
</aside>
<p>There’s something strange here, you mention the path being <code> C:/ShreyasBKR_Project/.git/</code>, but the presence of a <code>.git/</code> folder indicates that <code>ShreyasBKR_Project/</code> is already a repository.</p>
<aside class="quote no-group" data-username="bharatbondada" data-post="3" data-topic="190223">
<div class="title">
<div class="quote-controls"></div>
 bharatbondada:</div>
<blockquote>
<p>But unable to view the .git folder.</p>
</blockquote>
</aside>
<p>But you’re mentioning it in the path. That’s the Git folder.</p>
<p>You should try this: navigate in Bash or CMD to <code>C:/ShreyasBKR_Project/</code> and type:</p>
<pre><code>git status
</code></pre>
<p>this will give you either information about the repository status, or an error that it’s not a repository folder.</p>
<aside class="quote no-group" data-username="bharatbondada" data-post="3" data-topic="190223">
<div class="title">
<div class="quote-controls"></div>
 bharatbondada:</div>
<blockquote>
<p>I have checked for invisible folders also.</p>
</blockquote>
</aside>
<p>It looks like you’re working under Windows (always mention your OS when asking for help!), so yes you need to configure File explorer to show hidden and system folders if you want to see the contents of the <code>.git/</code> folder!</p>
<p>But I advise you to not tamper (nor worry) with the <code>.git/</code> folder, since its contents are handled by Git behind the scenes. You only need to access it for advanced stuff, and you need to know well what your doing.</p>

**Reply by Vicky8180** [reply]
I have also faced the same issue and tried everything you mentioned above and still facing the same problem. Can not able to push project.

**Reply by tajmone** [reply]
The original poster seems to be using MS Windows (as the "C:/" drive letter indicates).

Be aware that the `.git/` folder is a hidden folder and that Windows File Explorer by default hides hidden files and folders. Ensure that you enable showing hidden files and folders in the File Explorer settings.

Another factor to be aware of under Windows is that many folders under the installation partition (usually "C:/") are considered protected folders by the OS, which means that applications don't have permission to write into these folders. Windows has gradually increases the number of such protected folders, and now include the Desktop folder too. As a rule of thumb, I prefer to separate the installation and work space, and partition my disk so that C:/ is reserved for the OS and installed applications, and I use the other partition (usually "D:/") to store all my work files, projects, media, etc.

> I have also faced the same issue and tried everything you mentioned above and still facing the same problem. Can not able to push project.

You mention being unable to push to the repository, so I'm assuming you were able to create/clone the repository and that you do have a  `.git/` folder inside your project.

There could be many different reasons why you can't push to a repository, so it would be helpful if you could paste here the error you're getting in the terminal (Bash or CMD) so that we can see what the problem is.

Just to be clear, in order to push from a local repository to a remote repository you need to have created the remote repository first, and if you're planning to contribute to a third party repository then you need to first create on GitHub your own fork of the repository, then locally clone your fork, otherwise you won't have access permission to write on the upstream (original) repository, unless you are registered as a collaborator to that repository.
